docs/style-guide.md

```diff
@@ -0,0 +1,612 @@
+  this.dropdown = function(dropdownName) {
+    /**
+     * Dropdown api. Used to click on an element under a dropdown.
+     * @param {string} dropdownName
+     * @return {{option: Function}}
+     */
+    var openDropdown = function() {
+      element(by.css('.navbar-nav'))
+          .element(by.linkText(dropdownName))
+          .click();
+    };
+
+    return {
+      /**
+       * Get an option element under a dropdown.
+       * @param {string} optionName
+       * @return {ElementFinder}
+       */
+      option: function(optionName) {
+        openDropdown();
+        return element(by.css('.dropdown.open'))
+            .element(by.linkText(optionName));
+      }
+    }
+  };
+};
+
+module.exports = MenuPage;
+```
+
+```js
+var Menu = require('./menu');
+
+describe('protractor website', function() {
+
+  var menu = new Menu();
+
+  it('should navigate to API view', function() {
+    browser.get('http://www.protractortest.org/#/');
+
+    menu.dropdown('Reference').option('Protractor API').click();
+
+    expect(browser.getCurrentUrl())
+        .toBe('http://www.protractortest.org/#/api');
+  });
+});
+```
+
+**Why?**
+* When you have a large team and multiple e2e tests people tend to write
+  their own custom locators for the same directives.
+
+### Avoid text locators for text that changes frequently
+
+* Try to avoid text-based locators such as `by.linkText`, `by.buttonText`,
+  `by.cssContainingText`.
+
+**Why?**
+* Text for buttons, links, and labels tends to change over time. Minor text
+  changes in your application should not break your tests.
+
+# Test suites
+
+### Make your tests independent at least at the file level
+
+* Protractor can run your tests in parallel when you enable sharding. The files
+  are executed across different browsers as they become available.
+* Make your tests independent at the file level because the order in which
+  they run is not guaranteed and it's easier to run a test in isolation.
+
+### Do not add logic to your test
+
+* Avoid using if statements and for loops. When you add logic your test may
+  pass without testing anything, or may run very slowly.
+
+### Don't mock unless you need to
+
+This rule is a bit controversial, in the sense that opinions are very divided 
+when it comes to what the best practice is. Some developers argue that e2e 
+tests should use mocks for everything in order to avoid external network calls 
+and have a second set of integration tests to test the APIs and database. Other 
+developers argue that e2e tests should operate on the entire system and be as 
+close to the 'real deal' as possible.
+
+**Why?**
+* Using the real application with all the dependencies gives you high confidence
+* Helps you spot some corner cases you might have overlooked
+
+### Use Jasmine2
+
+**Why?**
+* Jasmine is well documented
+* It is supported by Protractor out of the box
+* You can use `beforeAll` and `afterAll`
+
+### Make your tests independent at file level
+
+**Why?**
```

Comment:
There's "Make your tests independent at least at the file level" above, so we should consolidate these two rules.